sympy/vector/coordsysrect.py

```diff
@@ -1,19 +1,45 @@
+import types
+from sympy.simplify import simplify
 from sympy.core.basic import Basic
 from sympy.vector.scalar import BaseScalar
-from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol
+from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol, symbols, sqrt, Lambda, Dummy
 from sympy.core.compatibility import string_types, range
 from sympy.core.cache import cacheit
 from sympy.vector.orienters import (Orienter, AxisOrienter, BodyOrienter,
                                     SpaceOrienter, QuaternionOrienter)
-import sympy.vector
 
 
-class CoordSysCartesian(Basic):
+def build_instance_from_lambda(fun=lambda x, y, z: (x, y, z)):
```

Comment:
This is the function to calculate the Lamé parameters @jbbskinny , though they are returned in an unusual way.

I chose to return a _Lambda_, because it does not require to define symbols. An alternative would have been to return a triple of symbols, say _r, theta, phi_ and their expression, say `1, r**2, r**2*sin(theta)**2`.

Try it yourself: `build_instance_from_lambda(lambda r, theta, phi: (r*sin(phi)*cos(theta), r*sin(phi)*sin(theta), r*cos(phi)))`